src/pipeline/src/etl/transform/transformer/greptime.rs

```diff
@@ -363,6 +363,60 @@ fn calc_ts(p_ctx: &PipelineContext, values: &VrlValue) -> Result<Option<ValueDat
     }
 }
 
+/// Converts VRL values to Greptime rows with per-row table suffixes.
+/// # Returns
+/// A vector of (row, table_suffix) pairs. Single object input produces one row,
+/// array input produces one row per array element with its own table_suffix.
+///
+/// # Errors
+/// - `ArrayElementMustBeObject` if an array element is not an object
+/// - `TransformArrayElement` if transformation fails for a specific element
+pub(crate) fn values_to_rows(
+    schema_info: &mut SchemaInfo,
+    values: VrlValue,
+    pipeline_ctx: &PipelineContext<'_>,
+    row: Option<Vec<GreptimeValue>>,
+    need_calc_ts: bool,
+    tablesuffix_template: Option<&crate::tablesuffix::TableSuffixTemplate>,
+) -> Result<Vec<(Row, Option<String>)>> {
+    use snafu::ResultExt;
+
+    use crate::error::{ArrayElementMustBeObjectSnafu, TransformArrayElementSnafu};
+    use crate::etl::ctx_req::ContextOpt;
```

Comment:
I don't like this style, please move it to above out of the function.

Reply:
fixed in https://github.com/GreptimeTeam/greptimedb/pull/7342/commits/35b4d74b294bf3d9db48c68e3381bdaaf8eca41b